XpressAI/xircuits
# ⚡️Optimize xircuits workflow save file

# Description

This will optimize which metadata should changes when saving/detecting changes. I took the liberty to update all examples metadata so that it's easier to see which metadata changes after saving. 

Currently, it's not possible to change the layers( `nodesLayer` and `linksLayer`) as everytime xircuits deserialized(detecting changes), it create the layers by default so it's hard to control that metadata. However, we can still control the nodes and links inside the layers. This PR will only update the metadata based on what have changes on the canvas like node position change, adding a link and etc. To summarize, this optimization will only update necessary changes beside **the layers' id**.

## Pull Request Type


## Type of Change


# Tests

I recommended using **VScodes's Source Control** to see the metadata changes.

1. Sanity check
   1. Open any example
   2. Directly save xircuits 
   3. Open VSCode and click Source Control on left sidebar
   4. Click on the changed example's xircuits file 
   5. You should see only the layers' id have been changed (It's not possible to changed that for now)
2. Move a node and see the changes
   1. Open any example
   2. Move any node
   3. Save xircuits
   4. Open VSCode and click Source Control on left sidebar
   5. Click on the changed example's xircuits file 
   6. See what have changed.
   7. The changed metadata should be the links points' position, node's position and its ports position. Include the layers id.

## Tested on?
